subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/VariantMetadataRules.java

```diff
@@ -139,6 +148,11 @@ private ImmutableRules() {
             super(null);
         }
 
+        @Override
+        public void setVariantDerivationStrategy(VariantDerivationStrategy variantDerivationStrategy) {
+            throw new UnsupportedOperationException("You are probably to set the derivation strategy to something that wasn't supposed to be mutable");
```

Comment:
Looks like there is a missing verb in that sentence.

Reply:
Good catch!